src/watts/plugin_dakota.py

```diff
@@ -0,0 +1,251 @@
+                output_data[name] = np.array(df[name])
+
+        # Save Dakota's final output '.dat' files
+        if (Path('finaldata1.dat').exists()):
+            with open('finaldata1.dat') as fd:
+                reader = csv.reader(fd)
+                rows = [row for idx, row in enumerate(reader) if idx == 0]
+
+            new_rows = str(rows[0][0]).split()
+            output_data['finaldata1'] = np.array([float(i) for i in new_rows])
+
+        return output_data
+
+
+class PluginDakota(TemplatePlugin):
+    """Plugin for running Dakota
+
+    Parameters
+    ----------
+    template_file
+        Templated Dakota input
+    extra_inputs
+        List of extra (non-templated) input files that are needed
+    extra_template_inputs
+        Extra templated input files
+    show_stdout
+        Whether to display output from stdout when Dakota is run
+    show_stderr
+        Whether to display output from stderr when Dakota is run
+
+    """
+
+    def __init__(self, template_file: str,  
+                 extra_inputs: Optional[List[str]] = None,
+                 extra_template_inputs: Optional[List[PathLike]] = None,
+                 auto_link_files: Optional[str] = None,
+                 show_stdout: bool = False, show_stderr: bool = False):
+        super().__init__(template_file, extra_inputs, extra_template_inputs,
+                         show_stdout, show_stderr)
+
+        dakota_dir = Path(os.environ.get("DAKOTA_DIR", ""))
+        self._executable = dakota_dir / f"dakota.sh"
+        self.input_name = template_file
+        self._auto_link_files = auto_link_files
+
+        # Setup to automatically include all 'extra_inputs' and 'extra_template_inputs' 
+        # to Dakota's "link files" option. Create a string of the file names in
+        # 'extra_inputs' and 'extra_template_inputs'.
+        if self._auto_link_files is not None:
+            dakota_link_files = []
+            if extra_inputs is not None:
+                dakota_link_files.extend(list(extra_inputs))
+
+            if extra_template_inputs is not None:
+                dakota_link_files.extend(list(extra_template_inputs))
+
+            self.dakota_link_files_string = " ".join(f"'{item}'" for item in dakota_link_files)
+
+    def prerun(self, params: Parameters, filename: Optional[str] = None):
+        """ Change the permisison of the Dakota driver file
+
+        Parameters
+        ----------
+        params
+            Parameters used to render template
+        """
+
+        # Store the string for the "link_files" in params.
+        if self._auto_link_files is not None:
+            params[self._auto_link_files] = self.dakota_link_files_string
+
+        super().prerun(params)
+        if 'dakota_driver_name' in params.keys():
+            os.chmod(params['dakota_driver_name'], 0o755)
+
+    @property
+    def execute_command(self):
+        return [str(self.executable), "-i", self.input_name]
+
+def run_dakota_driver(coupled_code_exec: str):
```

Comment:
Per PEP8, top-level classes/functions should have two blank lines between them
```suggestion


def run_dakota_driver(coupled_code_exec: str):
```